test/integration/controllermanager/seed/extensionscheck/extensionscheck_test.go

```diff
@@ -0,0 +1,265 @@
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	gardencorev1beta1helper "github.com/gardener/gardener/pkg/apis/core/v1beta1/helper"
+	"github.com/gardener/gardener/pkg/controllermanager/apis/config"
+	"github.com/gardener/gardener/pkg/controllermanager/controller/seed"
+	"github.com/gardener/gardener/pkg/controllerutils/mapper"
+	"github.com/gardener/gardener/pkg/utils/test"
+)
+
+const (
+	conditionThreshold = 2 * time.Second
+	syncPeriod         = 1 * time.Millisecond
+)
+
+var _ = Describe("Seed ExtensionsCheck controller tests", func() {
+	var (
+		seed *gardencorev1beta1.Seed
+		ci1  *gardencorev1beta1.ControllerInstallation
+		ci2  *gardencorev1beta1.ControllerInstallation
+	)
+
+	BeforeEach(func() {
+		By("creating Seed")
+		seed = &gardencorev1beta1.Seed{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: testID + "-",
+			},
+			Spec: gardencorev1beta1.SeedSpec{
+				Provider: gardencorev1beta1.SeedProvider{
+					Region: "region",
+					Type:   "providerType",
+				},
+				Settings: &gardencorev1beta1.SeedSettings{
+					ShootDNS:   &gardencorev1beta1.SeedSettingShootDNS{Enabled: true},
+					Scheduling: &gardencorev1beta1.SeedSettingScheduling{Visible: true},
+				},
+				Networks: gardencorev1beta1.SeedNetworks{
+					Pods:     "10.0.0.0/16",
+					Services: "10.1.0.0/16",
+					Nodes:    pointer.String("10.2.0.0/16"),
+					ShootDefaults: &gardencorev1beta1.ShootNetworks{
+						Pods:     pointer.String("100.128.0.0/11"),
+						Services: pointer.String("100.72.0.0/13"),
+					},
+				},
+				DNS: gardencorev1beta1.SeedDNS{
+					IngressDomain: pointer.String("someingress.example.com"),
+				},
+			},
+		}
+		Expect(testClient.Create(ctx, seed)).To(Succeed())
+		By(fmt.Sprintf("seed name is %s", seed.Name))
+		DeferCleanup(func() {
+			By("Delete Seed")
+			Expect(client.IgnoreNotFound(testClient.Delete(ctx, seed))).To(Succeed())
+		})
+
+		By("creating ControllerInstallations")
+		ci1 = &gardencorev1beta1.ControllerInstallation{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "foo-1",
+			},
+			Spec: gardencorev1beta1.ControllerInstallationSpec{
+				SeedRef: corev1.ObjectReference{
+					Name: seed.Name,
+				},
+				RegistrationRef: corev1.ObjectReference{
+					Name: "foo-registration",
+				},
+				DeploymentRef: &corev1.ObjectReference{
+					Name: "foo-deployment",
+				},
+			},
+		}
+
+		ci2 = ci1.DeepCopy()
+		ci2.SetName("foo-2")
+
+		//This is required so that the ExtensionsReady condition is created with appropriate lastUpdateTimestamp and lastTransitionTimestamp.
+		cleanUpTestVars := test.WithVars(
+			&gardencorev1beta1helper.Now, func() metav1.Time { return metav1.Time{Time: fakeClock.Now()} },
+		)
+		DeferCleanup(func() {
+			cleanUpTestVars()
+		})
```

Comment:
```suggestion
		// This is required so that the ExtensionsReady condition is created with appropriate lastUpdateTimestamp and lastTransitionTimestamp.
		DeferCleanup(test.WithVars(
			&gardencorev1beta1helper.Now, func() metav1.Time { return metav1.Time{Time: fakeClock.Now()} },
		))
```